### ISSUE TITLE
Report mass shift as a variable mod for pepxml import to Skyline

### ISSUE THREAD

**Issue by dkueltz** [Nesvilab/FragPipe#591]
Hi Alexey, Fangchao, Daniel,

I am using FragPipe to generate annotated spectra for import into Skyline and making raw spectral libraries for DIA.

 When using Open search and Mass offset common ptms search configurations in FragPipe I see many interesting PTMs and get about twice as many unique PSMs as with the Default configuration. In the Default configuration I fill up the Fragger PTM table with the max number of PTMs but that number is still not sufficient to accommodate most PTMs that are present. 

The advantage of using the Default Fragger config is that I can import the corresponding interact.pexml files into Skyline and all PTMs are properly recognized. When importing Open and Mass offset common PTMs configuration results I can also import the pepxmls into Skyline but none of the PTMs are showing up that are indicated in the csv tables generated by Fragpipe.

So for now I am limited to choosing a small set of the most interesting PTMs detected by the Open and Mass offset common PTM searches for Default searches.

I tried setting Open Search options - Report mass shift as a variable mod to Yes, keep delta mass and Yes, remove delta mass but this still does not work. I noticed that there is a note saying that these latter two options do not work with PTMshepard and suspect that this is a current limitation, which causes the issue with PTMs not being recognized in the pepxml by Skyline but I am not sure. I tried checking localize mass shifts and this gave results (I can see the PTM localizations in the csv tables) but does not change the pepxml behavior.

In short, it would be nice to be able to generate pepxml with Open and Mass shift common PTM search configurations that localize each PTM to a residue and then annotate it as a PTM with the corresponding mass shift in the pepxml files such that these PTMs can then be recognizes by Skyline just as they are for pepxml generated from Default searches

Thanks for any feedback you may be able to provide on this.
Best,
Dietmar

**Comment by anesvi** [comment]
Thanks. We added to the list of think we can try to implement

**Comment by dkueltz** [comment]
Hi Alexey,
Thank you for trying to address this request. I realize that this is not a trivial request as some (perhaps many?) of the delta masses may actually be the sum of multiple PTMs on different amino acids of the same peptide.
Dietmar

**Comment by dkueltz** [comment]
Hi Fengchao,

That would be a great feature and I am happy to try it. I have no problem using Fragpipe with Skyline through the interact.xml files but happy to try the new functionality.

Best,
Dietmar
_____________________________
Dietmar Kültz, PhD
Professor of Physiological Genomics
Department of Animal Sciences
University of California Davis
One Shields Ave., Meyer Hall
Davis, CA 95616, US
Phone: 530-752-2991 · Fax: 530-752-0175
The Kültz Lab @ PanoramaWeb<https://protect2.fireeye.com/v1/url?k=31323334-501d0a38-31357b2d-454441504e31-7aa348ecb6296162&q=1&e=190f4edb-9606-43a5-9557-955a1cea066b&u=https%3A%2F%2Fnam12.safelinks.protection.outlook.com%2F%3Furl%3Dhttps%253A%252F%252Fpanoramaweb.org%252FKueltz%252520Lab%252520-%252520UCDavis%26data%3D04%257C01%257Cgunter.wagner%2540yale.edu%257C373c964ab05a429feb2b08da0635b180%257Cdd8cbebb21394df8b4114e3e87abeb5c%257C0%257C0%257C637829124931649544%257CUnknown%257CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%253D%257C3000%26sdata%3DNlqRlS8EaoEOq9BpLgx3VlYapxbU1TZYu2nEXERlzjo%253D%26reserved%3D0>
Stress-induced Evolution · Targeted DIA Proteomics
Book: A Primer of Ecological Aquaculture (OUP 2022)<https://global.oup.com/academic/product/a-primer-of-ecological-aquaculture-9780198850236?facet_narrowbyvoicing_facet=Men%27s%20Voices&facet_narrowbyavailableforhire_facet=For%20Sale&type=listing&lang=en&cc=gb>

**Comment by fcyu** [comment]
Sounds great! Could you send an email to yufeATumich.edu (replace `AT` with `@`) to get the download links?

Thanks,

Fengchao

**Comment by dkueltz** [comment]
Hi Fengchao,

I just send the email and will download shortly after receiving the link. I let you know how it works for me.

Thanks!
Dietmar

**Comment by fcyu** [comment]
Thanks for letting me know. I have replied your email.

Best,

Fengchao

**Comment by dkueltz** [comment]
Got it! Downloading the files now.
Thanks,
Dietmar
